### PR TITLE
nginx: Address CVE-2024-7347

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Patch CVE 2024-7347

###### Change Log  

- Patch CVE 2024-7347

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-7347

###### Test Methodology

- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=625081&view=results

### CODE REVIEW

File: SPECS/nginx/nginx.spec [hunk 1]
```diff
@@ -18,6 +18,8 @@ Source1:        nginx.service
 Source2:        https://github.com/nginx/njs/archive/refs/tags/%{njs_version}.tar.gz#/%{name}-njs-%{njs_version}.tar.gz
 Source3:        https://github.com/open-telemetry/opentelemetry-cpp-contrib/archive/%{opentelemetry_cpp_contrib_git_commit}.tar.gz#/opentelemetry-cpp-contrib-%{opentelemetry_cpp_contrib_git_commit}.tar.gz
 Patch0:         CVE-2023-44487.patch
+Patch1:         CVE-2024-7347.1.patch
```
Comment (jslobodzian) [inline SPECS/nginx/nginx.spec:21]: I'm not sure that our tooling can handle the .1 and .2
Consider combining or dropping the .1 in the first patch and renaming the second patch something else?

Comment (jslobodzian) [reply SPECS/nginx/nginx.spec:21]: I have reached out the Karim to see if the CVE scanning tools can survive this.

Comment (kimoantiqe) [reply SPECS/nginx/nginx.spec:21]: Is there an explicit reason why we are dividing this .patch file into 2 parts? Why can't we have it in a single patch file?

Comment (kimoantiqe) [reply SPECS/nginx/nginx.spec:21]: @Camelron can you combine them into 1 patch file?

Comment (Camelron) [reply SPECS/nginx/nginx.spec]: Addressed, please see new patchfile.

**Pull request merged**